internal/controller/cilium_flow_collector.go

```diff
@@ -25,48 +24,62 @@ type CiliumFlowCollector struct {
 
 const (
 	ciliumHubbleRelayMaxFlowCount uint64 = 100
-	ciliumHubbleRelayServiceName  string = "hubble-relay"
-)
 
-var (
-	ErrHubbleNotFound   = errors.New("hubble Relay service not found; disabling Cilium flow collection")
-	ErrNoPortsAvailable = errors.New("hubble Relay service has no ports; disabling Cilium flow collection")
+	// Constants for fetching the mTLS secret
+	ciliumHubbleMTLSSecretName string = "hubble-relay-client-certs"
+	ciliumHubbleRelayNamespace string = "kube-system"
 )
 
-// discoverCiliumHubbleRelayAddress uses a kubernetes clientset in order to discover the address of the hubble-relay service within kube-system.
-func discoverCiliumHubbleRelayAddress(ctx context.Context, ciliumNamespace string, clientset kubernetes.Interface) (string, error) {
-	service, err := clientset.CoreV1().Services(ciliumNamespace).Get(ctx, ciliumHubbleRelayServiceName, metav1.GetOptions{})
+// newCiliumFlowCollector connects to Cilium Hubble Relay, sets up an Observer client, and returns a new Collector using it.
+func newCiliumFlowCollector(ctx context.Context, logger *zap.Logger, ciliumNamespace string) (*CiliumFlowCollector, error) {
+	clientset, err := NewClientSet()
 	if err != nil {
-		return "", ErrHubbleNotFound
+		return nil, fmt.Errorf("failed to create new client set: %w", err)
 	}
 
-	if len(service.Spec.Ports) == 0 {
-		return "", ErrNoPortsAvailable
+	service, err := hubble.DiscoverCiliumHubbleRelay(ctx, ciliumNamespace, clientset)
+	if err != nil {
+		return nil, fmt.Errorf("failed to discover Cilium Hubble Relay service: %w", err)
 	}
 
-	address := fmt.Sprintf("%s:%d", service.Spec.ClusterIP, service.Spec.Ports[0].Port)
-	return address, nil
-}
-
-// newCiliumCollector connects to Ciilium Hubble Relay, sets up an Observer client, and returns a new Collector using it.
-func newCiliumFlowCollector(ctx context.Context, logger *zap.Logger, ciliumNamespace string) (*CiliumFlowCollector, error) {
-	config, err := NewClientSet()
+	hubbleAddress, err := hubble.GetAddressFromService(service)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create new client set: %w", err)
+		return nil, fmt.Errorf("failed to discover Cilium Hubble Relay address: %w", err)
 	}
-	hubbleAddress, err := discoverCiliumHubbleRelayAddress(ctx, ciliumNamespace, config)
-	if err != nil {
-		return nil, err
+	isAKSManaged := aks.IsAKSManagedHubble(service.Annotations)
+
+	var conn *grpc.ClientConn
+	if isAKSManaged {
+		logger.Debug("Attempting mTLS connection for AKS-managed Cilium Hubble Relay", zap.String("address", hubbleAddress))
```

Comment:
There's nothing specific to AKS here.
```suggestion
		logger.Debug("Attempting mTLS connection to Cilium Hubble Relay", zap.String("address", hubbleAddress))
```